### PR TITLE
(DOCSP-21698): New Sync Hooks @realm/react docs

### PR DESCRIPTION
**Pull request by mongomoe**
### Jira

- https://jira.mongodb.org/browse/DOCSP-21698

### Staged Changes (Requires MongoDB Corp SSO)

- [Use Realm React](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-21698-realm-react-syncHooks/sdk/react-native/use-realm-react/) # Note: the main thing I need reviewed is from the "Using the App Provider" section and below.

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-realm/blob/master/REVIEWING.md)

### CODE REVIEW

File: source/sdk/react-native/use-realm-react.txt [hunk 1]
```diff
@@ -225,9 +225,72 @@ of a `FlatList <https://reactnative.dev/docs/flatlist>`_ component.
...
+
+.. _react-native-user-provider:
+
+Using the User Provider
+-----------------------
+
+The ``UserProvider`` automatically populates the logged-in user into the
+underlying :js-sdk:`SyncConfiguration <Realm.App.Sync.html#~SyncConfiguration>`.
+
+The UserProvider contains a fallback property that renders a Login component if
```
Comment (mongodben) [inline source/sdk/react-native/use-realm-react.txt:264]:
```suggestion
The ``UserProvider`` contains a fallback property that renders a Login component if
```

File: source/sdk/react-native/use-realm-react.txt [hunk 2]
```diff
@@ -225,9 +225,72 @@ of a `FlatList <https://reactnative.dev/docs/flatlist>`_ component.
...
+the user has not logged in yet. Once the user authenticates, Realm React renders
+the child components, and the application falls back to the Login component if
+the user logs out.
```
Comment (mongodben) [inline source/sdk/react-native/use-realm-react.txt:267]: i'd like more information on this `fallback` component. like will the application only render this component and none of the children of the `<UserProvider>`? could be a bit more explicit on what's happening here

File: source/sdk/react-native/use-realm-react.txt [hunk 3]
```diff
@@ -225,9 +225,72 @@ of a `FlatList <https://reactnative.dev/docs/flatlist>`_ component.
...
+    :language: typescript
+    :emphasize-lines: 1, 7, 11
+
+.. _react-native-user-provider:
+
+Using the User Provider
+-----------------------
+
+The ``UserProvider`` automatically populates the logged-in user into the
+underlying :js-sdk:`SyncConfiguration <Realm.App.Sync.html#~SyncConfiguration>`.
```
Comment (mongodben) [inline source/sdk/react-native/use-realm-react.txt:262]: can you provide an example of how "UserProvider automatically populates the logged-in user into the underlying SyncConfiguration" works? not clear to me

File: source/sdk/react-native/use-realm-react.txt [hunk 4]
```diff
@@ -225,9 +225,72 @@ of a `FlatList <https://reactnative.dev/docs/flatlist>`_ component.
    :ref:`sorted <react-native-sort-results>` list of tasks, read the :ref:`Read
    & Write Data <react-native-read-and-write-data>` docs.
 
+.. _react-native-app-provider:
+
+Using the App Provider
+----------------------
```
Comment (mongodben) [inline source/sdk/react-native/use-realm-react.txt:231]: i'd consider renaming the section titles. looks kinda funny with everything starting with 'Use/Using', without providing value to user. instead perhaps name the sections the name of their respect components/hooks in a more reference-type style.

i'd also consider making the hooks subsections of the providers that they're dependent upon. 

for ex: 
|__ AppProvider (H2)
|____ useApp (H3)
|__ UserProvider (H2)
|____ useUser (H3)

File: source/sdk/react-native/use-realm-react.txt (hunk 3 above)
Comment (mongomoe) [reply source/sdk/react-native/use-realm-react.txt]: Normally, you would have to do:
```
<RealmProvider sync={{partitionValue: 'SamplePartition', user: someUser}}>...</RealmProvider>
```

with the UserProvider the user is inferred, so you don't need to manually specify:

```
<UserProvider fallback={LoginComponent}>
     <RealmProvider sync={{partitionValue: 'SamplePartition'}}> ... </RealmProvider>
</UserProvider>
```

altered the text a bit (actually went back & forth on this paragraph quite a bit) lmk what you think.

File: source/sdk/react-native/use-realm-react.txt (hunk 2 above)
Comment (mongomoe) [reply source/sdk/react-native/use-realm-react.txt]: Yup only the component set as the fallback component is rendered when the user is *not* logged in. So if the user is not logged in -> render the fallback (i.e the Login component). If the user is logged in -> render the rest of the app.

File: source/sdk/react-native/use-realm-react.txt (hunk 4 above)
Comment (mongomoe) [reply source/sdk/react-native/use-realm-react.txt:231]: Whoops forgot to respond to this one, I did this for the AppProvider + UserProvider, the RealmProvider had a lot more content so its similarish but not exactly the same.

File: source/sdk/react-native/use-realm-react.txt [hunk 5]
```diff
@@ -114,7 +114,7 @@ of your application, with the ``RealmProvider`` component included in the
    For simple use-cases, you may want to wrap your entire application in the
    ``RealmProvider`` component, such as the example below. For additional security,
    you may only want to give access to the opened {+realm+} to specific screens, or
-   after the user has logged in.
+   after the user has logged-in.
 
 Usage
 ~~~~~
```
Comment (mongodben) [inline source/sdk/react-native/use-realm-react.txt:120]: nit: consider removing the subheading "usage". looks a little funny w heading 'usage' right under 'using the realm provider

File: source/sdk/react-native/use-realm-react.txt [hunk 6]
```diff
@@ -151,15 +151,13 @@ waiting for {+sync+} to open:
 .. literalinclude:: /examples/generated/expo/AppWrapper.snippet.dynamically-update-realm-config.tsx
     :language: typescript
```
[the remark this answers is not in the document]
Comment (mongodben) [reply source/sdk/react-native/use-realm-react.txt:152]: i think your other changes rendered this comment null. closing out.

**Pull request merged**